builtin/logical/transit/policy.go

```diff
@@ -335,11 +358,27 @@ func (p *Policy) DeriveKey(context []byte, ver int) ([]byte, error) {
 		return nil, errutil.UserError{Err: "missing 'context' for key deriviation. The key was created using a derived key, which means additional, per-request information must be included in order to encrypt or decrypt information"}
 	}
 
-	switch p.KDFMode {
-	case kdfMode:
+	switch p.KDF {
```

Comment:
We don't -- any time the key is loaded it will be checked for needing an upgrade. But in the latest commit we actually don't need an upgrade. We just drop the `KDFMode` value from `Policy` in favor of the `KDF` value whose default (`0`) matches the const identifier of the initial KDF. So that upgrade will be seamless.